src/pages/achievement/control/AchievementControl.tsx

```diff
@@ -35,8 +35,18 @@ const AchievementControl: React.FC = () => {
     [dispatch]
   );
 
-  const inferencer = useTypedSelector(
-    state => new AchievementInferencer(state.achievement.achievements, state.achievement.goals)
+  // TODO: This is a hacky fix. By right, we shouldn't need to use an
+  // inferencer instance since we can encapsulate the logic using hooks
+  // and component state.
+  const [initialAchievements, initialGoals] = useTypedSelector(state => [
+    state.achievement.achievements,
+    state.achievement.goals
+  ]);
```

Comment:
`useTypedSelector` automatically listens to any state changes and reruns the callback, resulting in a new instance of the inferencer created each time the state is updated. This led to the bugged behaviour where it could not keep track of the state of the newly added (unpublished) goals and achievements.